internal/extract_test.go

```diff
@@ -60,17 +65,152 @@ func TestTar(t *testing.T) {
-//	defer utility.LoggedClose(result, "")
-//	var crypter internal.OpenPGPCrypter
-//	err = internal.DecryptAndDecompressTar(result, readerMaker, &crypter)
-//	assert.NoError(t, err)
-//}
+func noPassphrase() (string, bool) {
+	return "", false
+}
+
+func TestDecryptAndDecompressTar_unencrypted(t *testing.T) {
+	sb := testtools.NewStrideByteReader(10)
+	in := &io.LimitedReader{
+		R: sb,
+		N: int64(1024),
+	}
+	b, err := ioutil.ReadAll(in)
+	bCopy := make([]byte, len(b))
+	copy(bCopy, b)
+
+	compressor := GetLz4Compressor()
+	compressed := internal.CompressAndEncrypt(bytes.NewReader(b), compressor, nil)
+
+	compressedBuffer := &bytes.Buffer{}
+	_, _ = compressedBuffer.ReadFrom(compressed)
+	brm := &BufferReaderMaker{compressedBuffer, "/usr/local/test.tar.lz4"}
+
+	decompressed := &bytes.Buffer{}
+	err = internal.DecryptAndDecompressTar(decompressed, brm, nil)
+
+	if err != nil {
+		t.Logf("%+v\n", err)
+	}
+	assert.Equalf(t, bCopy, decompressed.Bytes(), "decompressed tar does not match the input")
+}
+
+func TestDecryptAndDecompressTar_encrypted(t *testing.T) {
+	sb := testtools.NewStrideByteReader(10)
+	in := &io.LimitedReader{
+		R: sb,
+		N: int64(1024),
+	}
+
+	b, err := ioutil.ReadAll(in)
+
+	// Copy generated bytes to another slice to make the test more robust against modifications of "b".
+	bCopy := make([]byte, len(b))
+	copy(bCopy, b)
+
+	crypter := openpgp.CrypterFromKeyPath(PrivateKeyFilePath, noPassphrase)
+
+	compressor := GetLz4Compressor()
+	compressed := internal.CompressAndEncrypt(bytes.NewReader(b), compressor, crypter)
+
+	compressedBuffer, err := ioutil.ReadAll(compressed)
+	brm := &BufferReaderMaker{bytes.NewBuffer(compressedBuffer), "/usr/local/test.tar.lz4"}
+
+	decompressed := &bytes.Buffer{}
+	err = internal.DecryptAndDecompressTar(decompressed, brm, crypter)
+
+	if err != nil {
+		t.Logf("%+v\n", err)
+	}
+
+	assert.Equalf(t, bCopy, decompressed.Bytes(), "decompressed tar does not match the input")
+}
+
+func TestDecryptAndDecompressTar_noCrypter(t *testing.T) {
+	sb := testtools.NewStrideByteReader(10)
+	in := &io.LimitedReader{
+		R: sb,
+		N: int64(1024),
+	}
+
+	b, err := ioutil.ReadAll(in)
+
+	// Copy generated bytes to another slice to make the test more robust against modifications of "b".
+	bCopy := make([]byte, len(b))
+	copy(bCopy, b)
+
+	crypter := openpgp.CrypterFromKeyPath(PrivateKeyFilePath, noPassphrase)
+
+	compressor := GetLz4Compressor()
+	compressed := internal.CompressAndEncrypt(bytes.NewReader(b), compressor, crypter)
+
+	compressedBuffer, err := ioutil.ReadAll(compressed)
+	brm := &BufferReaderMaker{bytes.NewBuffer(compressedBuffer), "/usr/local/test.tar.lz4"}
+
+	decompressed := &bytes.Buffer{}
+	err = internal.DecryptAndDecompressTar(decompressed, brm, nil)
+
+	if err != nil {
+		t.Logf("%+v\n", err)
+	}
+
+	assert.Error(t, err)
+	assert.EqualError(t, err, "DecryptAndDecompressTar: lz4 decompress failed. Is archive encrypted?: DecompressLz4: lz4 write failed: invalid lz4 data")
```

Comment:
Instead of check message error in tests, it would be better to check the error type. And you can create new if you need it.

Reply:
And fix it in other tests too